### PR TITLE
fix: broadcast from tun to clients only sends to the last client

### PR DESCRIPTION
**Pull request by copy**

### CODE REVIEW

**Comment by copy** [comment]
It seems this didn't cause as many problems as one would expect, since usually a broadcast was a response to the mostly recently connected client (e.g. dhcp offers are sent via broadcast). I found this bug while trying to connect to VMs at the same time.

(The next version of v86 will have packet capturing that can be imported in wireshark)

**Comment by benjamincburns** [comment]
> (The next version of v86 will have packet capturing that can be imported in wireshark)
Oh that sounds all kinds of useful!

And good on you for wading into this code. For such a small module it's horrendously unreadable, but I've been hesitant to change it much over the years because it's been so hecking stable. Like literally it was just sitting in a tmux session for the first 3ish years I was running it.

File: switchedrelay.py [hunk 1]
```diff
@@ -66,13 +67,13 @@ def run(self):
...
                                 for socket in macmap.values():
-                                    def send_message():
+                                    def send_message(socket):
                                         try:
                                             socket.rate_limited_downstream(str(buf))
                                         except:
                                             pass
 
-                                    loop.add_callback(send_message)
+                                    loop.add_callback(functools.partial(send_message, socket))
```
Comment (benjamincburns) [inline switchedrelay.py:76]: Hmm, I'm not sure if this change has any effect?

In the prior version `send_message` was closing over the `socket` variable.

In your version you've added a new `socket` positional arg to the `send_message` function that aliases out the `socket` from the `for socket in macmap.values()` iteration. But then when it's passed to `loop.add_callback` you're binding that arg back to the same `socket` variable from the for loop.

I think that's functionally identical?

Comment (benjamincburns) [reply switchedrelay.py:76]: Hmm, I might be confused over when the closed-over `socket` var in the original version actually binds. I'll do a quick comparison in a REPL to see what's what.

Comment (benjamincburns) [reply switchedrelay.py:76]: Ah, such a subtle bug... good catch!

```python
from functools import partial

vals = [1, 2, 3, 4, 5, 6]

first_run = []
second_run = []

for val in vals:
    def print_val():
        print(f"first-run: {val}")

    first_run.append(print_val)

for val2 in vals:
    def print_val(val2):
        print(f"second-run: {val2}")

    second_run.append(partial(print_val, val2))

[ a() for a in first_run ]
[ a() for a in second_run ]

```

Gives the following output:

```
first-run: 6
first-run: 6
first-run: 6
first-run: 6
first-run: 6
first-run: 6
second-run: 1
second-run: 2
second-run: 3
second-run: 4
second-run: 5
second-run: 6
```

**Pull request merged**